ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -536,6 +548,62 @@ private AdminResponse handleBlobIndexRequest(DataInputStream requestStream, Admi
         errorMessage.getBytes());
   }
 
+  /**
+   * Handles {@link ForceDeleteAdminRequest}.
+   * @param requestStream the serialized bytes of the request.
+   * @param adminRequest the {@link AdminRequest} received.
+   * @return the {@link AdminResponse} to the request.
+   */
+  private AdminResponse handleForceDeleteRequest(DataInputStream requestStream, AdminRequest adminRequest) {
+    final int correlationId = adminRequest.getCorrelationId();
+    final String clientId = adminRequest.getClientId();
+    // check if the force delete request is enabled.
+    if (!serverConfig.serverHandleForceDeleteRequestEnabled) {
+      logger.error("ForceDelete request is disabled");
+      return new AdminResponse(correlationId, clientId, ServerErrorCode.Bad_Request);
+    }
+
+    ForceDeleteAdminRequest forceDeleteAdminRequest;
+    try {
+      forceDeleteAdminRequest = ForceDeleteAdminRequest.readFrom(requestStream, adminRequest, storeKeyFactory);
+    } catch (Exception e) {
+      logger.error("Failed to deserialize ForceDeleteAdminRequest", e);
+      return new AdminResponse(correlationId, clientId, ServerErrorCode.Bad_Request);
+    }
+
+    StoreKey storeKey = forceDeleteAdminRequest.getStoreKey();
+    short lifeVersion = forceDeleteAdminRequest.getLifeVersion();
+    long operationTimeMs = System.currentTimeMillis();
+    BlobId blobId = (BlobId)storeKey;
+    short accountId = blobId.getAccountId();
+    short containerId = blobId.getContainerId();
+
+    try {
+      PartitionId partitionId = blobId.getPartition();
+      Store store = storeManager.getStore(partitionId);
+      if (store == null) {
+        logger.error("ForceDeleteAdminRequest BlobId {}'s partition[{}] doesn't exist or stopped in this host", blobId, partitionId.getId());
+        return new AdminResponse(correlationId, clientId, ServerErrorCode.Replica_Unavailable);
+      }
+      // Get the size of the delete record
+      MessageFormatInputStream stream =
+          new DeleteMessageFormatInputStream(storeKey, accountId, containerId, operationTimeMs, lifeVersion);
+      // Generate the MessageInfo
```

Comment:
we don't need to create a stream here, you can just pass a message info to the forceDelete method. We will recalculate the size in the forceDelete.

Reply:
Removed. Thanks,